extensions/positron-python/python_files/posit/positron/positron_ipkernel.py

```diff
@@ -499,6 +499,8 @@ def __init__(self, **kwargs) -> None:
             module="jedi",
         )
 
+        warnings.filterwarnings("once", category=DataExplorerWarning)
```

Comment:
Hm I think something about this is a little weird. I'm seeing multiple warnings that pile up next to each other (even if I've moved on with other code in the console). Here's a state diagram where every time I hit a pink box I get a new warning:

```mermaid
graph LR
a[create DF]-->b[click view button]:::warn
b-->c[click view button]
c-->c
a-->d[%view df]:::warn
d-->b
b-->d
d-->d
c-->d
classDef warn fill:#ff7272
```
I suppose most people would just install numpy the first time so they wouldn't hit this scenario, so not a big deal.

Reply:
Ah, turns out the "once" filter doesn't quite work right since we have our own showwarning 😞 I added a simple switch to turn off this type of warning after it's been displayed once to get around this!